### PR TITLE
feat(types): make `Model.init` aware of pre-configured foreign keys

### PR DESCRIPTION
**Pull request by ephys**
### Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] Does `yarn test` or `yarn test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

Closes https://github.com/sequelize/sequelize/issues/14125

`Model.init` currently requires a configuration for every declared attributes.

Some attributes don't need to be configured by `Model.init`, these include:
- Timestamp attributes.
- Foreign keys added by association methods (like `Project.belongsTo(User)`).

I've resolved this by adding a new `ForeignKey<>` branded type (we're now at 3 branded types).
This type tells `Model.init` that the attribute doesn't need to be included.

I didn't include a solution for timestamp attributes, because the 'workaround' (which has been documented) is as simple and clearer than adding another branded type.

---

I've also included some documentation updates from https://github.com/sequelize/sequelize/pull/14091 that I want to cherry-pick in v6.

### CODE REVIEW

File: test/types/infer-attributes.ts [hunk 1]
```diff
@@ -2,11 +2,12 @@ import { expectTypeOf } from 'expect-type';
 import {
   Attributes,
   CreationAttributes,
-  CreationOptional,
+  CreationOptional, DataTypes, ForeignKey,
```
Comment (WikiRik) [inline test/types/infer-attributes.ts:5]: Just for consistency can we move this over three lines?

File: test/types/infer-attributes.ts [hunk 2]
```diff
@@ -39,6 +41,15 @@ class User extends Model<InferAttributes<User, { omit: 'omittedAttribute' | 'omi
   }
 }
 
+User.init({
+  mandatoryArrayAttribute: DataTypes.ARRAY(DataTypes.STRING),
+  mandatoryAttribute: DataTypes.STRING,
+  // projectId is omitted but still works, because it is branded with 'ForeignKey'
```
Comment (WikiRik) [inline test/types/infer-attributes.ts:47]: Do we need this comment?

**Review by WikiRik — COMMENTED** [review]
Not knowledgeable enough to actually review the changes in src/model.d.ts but in general looks good. I am wondering however if we can't add some additional tests for this (apart from the one you've already added)

**Review by npereira-jc — DISMISSED** [review]

**Review by WikiRik — DISMISSED** [review]

**Comment by ephys** [comment]
@WikiRik resolving the conflict dismissed your review 😅

**Review by WikiRik — APPROVED** [review]

**Comment by ephys** [comment]
Thank you!

**Pull request merged**